### DISCUSSION TITLE
non-manifold mesh repair

### DISCUSSION THREAD

**Discussion by jigneshBadrakhiya** [elalish/manifold#572] [General]
non manifold mesh repair to manifold mesh bcoz boolean intersection error Error::NotManifold!

**Comment by pca006132** [comment]
I think this depends on what your mesh is. There is a [merge function](https://manifoldcad.org/docs/html/structmanifold_1_1_mesh_g_l.html#a2078d075bef279041b67769e4197c254) for `MeshGL` that tries to merge vertices in order to convert non-manifold mesh to manifold mesh in a best-effort manner.
If it is not enough, maybe you need to use some external tools for repair, e.g. meshlab.

**Reply by jigneshBadrakhiya** [reply]
apply merge function but error still on.
if (!IsManifold()) {
    MarkFailure(Error::NotManifold);
    return;
  }

**Reply by pca006132** [reply]
you will need to manually fix the mesh in this case. in general it is not possible to convert non-manifold mesh into manifold mesh.

**Reply by elalish** [reply]
Some of these discussions might be better to switch to Q&A and mark an answer, rather than just closing them. It might make it easier for people to find the answers when they have a similar problem.
